pkg/oci.go

```diff
@@ -74,48 +71,66 @@ func PushSBOM(sbomDoc *v2_3.Document, sbomDescriptor *v1.Descriptor, sbomBytes [
 		ManifestAnnotations: annotations,
 	})
 	if err != nil {
-		return fmt.Errorf("error packing manifest: %w", err)
+		return nil, fmt.Errorf("error packing manifest: %w", err)
 	}
 
 	// Use the latest tag if no tag is specified
 	tag := "latest"
 	ref, err := registry.ParseReference(reference)
 	if err != nil {
-		return fmt.Errorf("error parsing reference: %w", err)
+		return nil, fmt.Errorf("error parsing reference: %w", err)
 	}
 
 	if ref.Reference != "" {
 		tag = ref.Reference
 	}
 
 	if err = mem.Tag(ctx, manifestDescriptor, tag); err != nil {
-		return err
+		return nil, err
 	}
 
-	// Connect to a remote repository
-	repo, err := remote.NewRepository(reference)
+	// Copy from the memory store to the remote repository
+	manifest, err := oras.Copy(ctx, mem, tag, dest, tag, oras.DefaultCopyOptions)
+	return &manifest, err
+}
+
+// AttachArtifact attaches an artifact to the subject descriptor
+func AttachArtifact(subject *v1.Descriptor, artifactDescriptor *v1.Descriptor, artifactType string, artifactBytes []byte, dest oras.Target) (*v1.Descriptor, error) {
+	ctx := context.Background()
```

Comment:
Ideally `ctx` should be passed all the way in from outside for context cancel capability?

Reply:
Opened #18 to track this, will address in a separate PR